app/controllers/middleware_server_controller.rb

```diff
@@ -262,8 +273,16 @@ def run_server_operation(operation_info, items)
     operation_triggered = false
     items.split(/,/).each do |item|
       mw_server = identify_record item
-      if mw_server.product == 'Hawkular' && operation_info.fetch(:skip)
-        add_flash(_("Not %{hawkular_info} the provider") % {:hawkular_info => operation_info.fetch(:hawk)})
+      if skip_operation?(mw_server, operation_info)
+        message = if operation_info.key?(:skip_msg)
+                    operation_info.fetch(:skip_msg)
+                  else
+                    "Not %{operation_name} the provider itself"
```

Comment:
You need to get this string to the catalog -- use `N_()` on this line to get it done.

Although composing an English sentence from words will make i18n impossible. So you should rather have a list of messages for all the operations.